usl_pipeline/cloud_functions/main.py

```diff
@@ -307,12 +308,38 @@ def _read_wps_features(fd: IO[bytes]) -> Tuple[NDArray, FeatureMetadata]:
     # Ignore type checker error - BytesIO inherits from expected type BufferedIOBase
     # https://shorturl.at/lk4om
     with xarray.open_dataset(fd) as ds:  # type: ignore
-        feature = ds.data_vars["SNOALB"].values
-        # Drop time axis - each WPS output file will correspond to single time value
-        feature = numpy.squeeze(feature, axis=0)
+        features_matrix = numpy.array([])
+        for var in wps_variables.REQUIRED_VARS:
+            try:
+                feature = ds.data_vars[var].values
+                feature = _process_wps_feature(feature)
+
+                if not features_matrix.size:
+                    features_matrix = feature
+                else:
+                    features_matrix = numpy.dstack((features_matrix, feature))
+            except KeyError:
+                logging.warning(f"No variable named '{var}' found in dataset.")
 
     # TODO: Write to metastore
-    return feature, FeatureMetadata()
+    return features_matrix, FeatureMetadata()
+
+
+def _process_wps_feature(feature: NDArray) -> NDArray:
+    """Performs a series of array transforms on a WPS variable array.
+
+    1. Drops time axis - since each WPS output file corresponds to one datetime
+    2. TBD (feature scaling, extracting levels, etc)...
```

Comment:
FYI things like scaling will be different per feature type, so we'd need the feature name or such to dispatch behaviour against in here.

Reply:
Good call - updated to pass in the xarray dataarray object instead